### PR TITLE
rostopic: repeatedly republish message from file

### PR DESCRIPTION
**Pull request by cwecht**
This replaces #1348

### CODE REVIEW

**Comment by cwecht** [comment]
hm, looks like the build-infastructure had an hiccup ...

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

File: tools/rostopic/src/rostopic/__init__.py [hunk 1]
```diff
@@ -1945,6 +1945,13 @@ def stdin_publish(pub, msg_class, rate, once, filename, verbose):
     else:
         iterator = stdin_yaml_arg
 
+    exactly_one_message = False
+    for _ in iterator():
+        if exactly_one_message:
+            exactly_one_message = False
+            break
+        exactly_one_message = True
```
Comment (dirk-thomas) [inline tools/rostopic/src/rostopic/__init__.py]: Doesn't this alter the content returned by the iterator in the later loop - in the case of the iterator being `stdin_yaml_arg`?

Comment (cwecht) [reply tools/rostopic/src/rostopic/__init__.py]: Yes, that's right. The fix makes not much sense for stdin stream anyway, so I limited it to actual files now.

**Comment by cwecht** [comment]
flaky tests again...

**Comment by cwecht** [comment]
@ros-pull-request-builder retest this please

**Comment by dirk-thomas** [comment]
Thanks for the patch and for iterating on it.

Ignoring the flaky test on CI...

**Comment by pavloblindnology** [comment]
substitute_keywords argument isn't used in this scenario.

**Pull request merged**